app/src/lib/stores/app-store.ts

```diff
@@ -1529,7 +1528,7 @@ export class AppStore extends TypedBaseStore<IAppState> {
     }
 
     this.repositoryStateCache.update(repository, () => ({
-      conflictState: null,
+      changesState: { ...repoState.changesState, conflictState: null },
```

Comment:
As above, `this.repositoryStateCache.updateChangesState` can simplify this code.